internal/build/fix_whitespace.sh

```
@@ -0,0 +1,64 @@
+	[ -f "$file" ] || return 0
+
+	# Skip empty files
+	if [ ! -s "$file" ]; then
+		return 0
+	fi
+
+	# Remove trailing whitespace
+	sed -i 's/[[:space:]]*$//' "$file"
+
+	# Check if file became empty or single newline after sed
+	local size=$(stat -c%s "$file" 2>/dev/null || stat -f%z "$file" 2>/dev/null || echo 0)
```

Shellcheck (0.10.0)

[warning] In POSIX sh, 'local' is undefined.

(SC3043)